pkg/cache/cache.go

```diff
@@ -113,6 +116,11 @@ type Options struct {
 	// [1] https://pkg.go.dev/k8s.io/apimachinery/pkg/fields#Selector
 	// [2] https://pkg.go.dev/k8s.io/apimachinery/pkg/fields#Set
 	SelectorsByObject SelectorsByObject
+
+	// UnsafeDisableCacheDeepCopy indicates not to deep copy objects during get or list objects.
+	// Be very careful with this, when enabled you must DeepCopy any object before mutating it,
+	// otherwise you will mutate the object in the cache.
+	UnsafeDisableCacheDeepCopy *DisableCacheDeepCopy
```

Comment:
Can this just be a `bool` now?

Also name nit: since this is in the cache package on the `cache.Options` struct, I don't think we need the word `Cache` in the name.

```suggestion
	// UnsafeDisableDeepCopy indicates not to deep copy objects during get or list objects.
	// Be very careful with this, when enabled you must DeepCopy any object before mutating it,
	// otherwise you will mutate the object in the cache.
	UnsafeDisableDeepCopy bool
```

Reply:
> Can this just be a `bool` now?

I changed the type from `bool` to `struct` in case we might need to add some optional policies for it in future.

> Also name nit: since this is in the cache package on the cache.Options struct, I don't think we need the word Cache in the name.

Have removed `Cache` in the name.

Reply:
Not sure if @vincepri @alvaroaleman and @varshaprasad96 have opinions, but I think I'd prefer to make this a `bool` since we have no options right now to deal with (and may never).

If we _do_ end up needing to configure other aspects of this in the future, we could introduce an `UnsafeDisableDeepCopyOptions` struct alongside this `bool` if we wanted to avoid a breaking change.

Reply:
Alright, I have changed it back to `bool` 🤔